### DISCUSSION TITLE
Bottom Trigger key codes

### DISCUSSION THREAD

**Discussion by MihinMUD** [sezanzeb/input-remapper#799] [Q&A] [answered]
I am trying to map my keyboard to a joystick and I am struggling with the trigger keys. I was able to map the top 2 triggers with `BTN_TL` and `BTN_TR`. But the bottom 2 didn't work. I tried `BTN_TL2` and `BTN_TR2` but it didn't work for some reason. Maybe those 2 are analog and I have to do something with `event()`? If so what are the code numbers for them.

**Comment by jonasBoss** [comment, marked as answer]
`BTN_TL2` and `BTN_TR2` should work. Did you check with `sudo evtest` that that `BTN_TL2` and `BTN_TR2` are not sent?

Gamepads are in general a huge problem, it seems like every application has its own idea how a gamepad looks, so it is almost impossible to create one that works in all scenarios.

**Reply by MihinMUD** [reply]
Ah, I see. Yeah that could be it. `evtest` is detecting the triggers. It might be the game I'm playing thinking of something different. Thanks for the help. I'll test every BTN code to see what the game want!
